allennlp/commands/train.py

```diff
@@ -336,10 +318,6 @@ def _train_worker(
     master_port: int = 29500,
     world_size: int = 1,
     distributed_device_ids: List[str] = None,
```

Comment:
`recover` is unused here. I wonder why.

Reply:
I traced the logic as I was doing the `Lazy[]` refactoring, and I was pretty sure it was only used when creating the serialization directory (we check to see if it already exists).  There was a use in vocab extension, but the refactoring made that unnecessary.  I didn't do a thorough check of the `recover` pipeline, because I'm pretty sure we have tests that check it.

At this point I *think* this parameter is unnecessary for this function and can be removed.  If you want to check that and remove it in a subsequent PR, that would be awesome.